commonknowledge/mapped
# Add generate a map button to inspect external data source page

## Description
This PR adds the following:
A generate map for this data source button on the inspect external data source page that loads the data source onto a map and redirects the user to it
A check so that if a map with this data source already exists we provide a link to that map

## Motivation and Context
Addresses issue [MAP-354](https://linear.app/commonknowledge/issue/MAP-354/when-i-set-up-or-look-at-a-data-source-a-button-to-view-this-data)

## How Can It Be Tested?
Download the branch and run locally
Import an external data source
Click on the Generate map for this data source button and observe that it redirects to a map with this data source loaded
Try to click on the button a second time and observe that the toast message offers a link to the existing map

## How Will This Be Deployed?
Normal deployment process
